src/Tempest/Mapper/src/Casters/CasterFactory.php

```diff
@@ -48,6 +49,7 @@ public function forProperty(PropertyReflector $property): ?Caster
             'float' => new FloatCaster(),
             'bool' => new BooleanCaster(),
             DateTimeImmutable::class, DateTimeInterface::class, DateTime::class => DateTimeCaster::fromProperty($property),
+            Upload::class => new FileUploadCaster(),
```

Comment:
You should use the `#[CastWith]` attribute on the upload class. This mapping is only for classes and types that we don't control